db/migrate/20210527151451_add_lead_provider_to_participation_record.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class AddLeadProviderToParticipationRecord < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :participation_records, :lead_provider, null: true, index: true, foreign_key: true, type: :uuid
+    ParticipationRecord.all.each do |pr|
+      ect_profile = pr.early_career_teacher_profile
+      school_cohort = SchoolCohort.find_by(school: ect_profile.school, cohort: ect_profile.cohort)
+      pr.update!(lead_provider_id: school_cohort&.lead_provider.id)
```

Comment:
Probably want `school_cohort&.lead_provider&.id` (`&` before `id`) here.

Or (probably better) do something like
```
lead_provider = school_cohort&.lead_provider || LeadProvider.first
pr.update!(lead_provider_id: lead_provider.id
```
to handle the edge cases somehow.